Sources/GitBuddyCore/CommandRegistry.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  CommandRegistry.swift
+//  GitBuddyCore
+//
+//  Created by Antoine van der Lee on 03/02/2020.
+//
+
+import Foundation
+import SPMUtility
+import Basic
+
+/// Allows to register subcommands that can be run.
+struct CommandRegistry {
+
+    private let parser: ArgumentParser
+    private var commands: [Command] = []
+    private let arguments: [String]
+    private let environment: [String: String]
+
+    init(usage: String, overview: String, arguments: [String], environment: [String: String]) {
+        parser = ArgumentParser(usage: usage, overview: overview)
+        self.arguments = arguments
+        self.environment = environment
+    }
+
+    mutating func register(commandType: Command.Type) throws {
+        let command = try commandType.init(parser: parser)
+        commands.append(command)
+    }
+
+    private func processArguments() throws -> ArgumentParser.Result {
+        return try parser.parse(Array(arguments.dropFirst()))
```

Comment:
It could be good to add a comment explaining why we drop the first argument here.

Reply:
Yes, good point!